assets/src/components/account/AccountOverview.tsx

```diff
@@ -112,25 +118,90 @@ class AccountOverview extends React.Component<Props, State> {
     return <Table dataSource={data} columns={columns} />;
   };
 
+  handleChangeCompanyName = (e: any) => {
+    this.setState({
+      account: {
+        ...this.state.account,
+        company_name: e.target.value,
+      },
+    });
+  };
+
+  handleStartEditing = () => {
+    this.setState({isEditing: true});
```

Comment:
I think since we want to distinguish between what's showing up in the UI vs what's actually saved in the DB, it might be good to just add a `companyName` field to `this.state`, and then do something like set `this.setState({companyName: this.state.account.company_name})` here or whenever we get the `account` on mount, and then do a reset whenever the user hits "cancel"

one reason for this is if I'm editing the company name and then I hit cancel, I want to make sure the company name that is displayed in the UI goes back to whatever is actually saved

(this is how we currently handle it in the user profile: https://github.com/papercups-io/papercups/blob/master/assets/src/components/account/UserProfile.tsx#L88 basically if the user hits "cancel", we fetch the profile info from the server again and reset it... I imagine it would be good to do something similar here)

what do you think?

Reply:
Hi Alex, thank you for the feedback here! That all makes a lot of sense, and I’ve made the changes you suggested.

One note - I based my `fetchLatestAccountInfo` method on the `fetchLatestProfile` method from the user profile. I carried over the `else` branch and the related comment from the user profile. I’m familiar with optimistic vs. pessimistic updates, but I'm not quite clear on how these else branches are functioning.

I just wanted to flag that, and I’m happy to keep it or make any updates. If easier, I can jump on Slack to chat too.

Sorry for the long comment - wanted to get everything in one place for you :) Thanks!

**AccountOverview.tsx**
```
fetchLatestAccountInfo = async () => {
   const account = await API.fetchAccountInfo();
  
   if (account) {
     this.setState({account});
     this.setState({companyName: this.state.account.company_name});
   } else {
     // NB: this also handles resetting this value if the optimistic update fails
     this.setState({companyName: ''});
   }
 };
```

**UserProfile.tsx**
```
fetchLatestProfile = async () => {
    const profile = await API.fetchUserProfile();

    if (profile) {
      const {
        display_name: displayName,
        full_name: fullName,
        profile_photo_url: profilePhotoUrl,
      } = profile;

      this.setState({
        displayName,
        fullName,
        profilePhotoUrl,
      });
    } else {
      // NB: this also handles resetting these values if the optimistic update fails
      this.setState({
        displayName: '',
        fullName: '',
        profilePhotoUrl: '',
      });
    }
  };
```

Reply:
ah yes, sorry for the confusion 😅  i tried to explain in the comments below but let me know if anything doesn't make sense!

Reply:
No worries and thank you for taking the time to explain! Definitely all makes sense now. I think we should be good to go :)